crates/pow/Cargo.toml

```diff
@@ -15,10 +15,10 @@ version.workspace      = true
 workspace = true
 
 [dependencies]
-serde     = { workspace = true }
-sha2      = { workspace = true }
-thiserror = { workspace = true }
-tokio     = { features = ["fs"], workspace = true }
+miden-client = { workspace = true }
+sha2         = { workspace = true }
+thiserror    = { workspace = true }
+tokio        = { features = ["time"], workspace = true }
```

Comment:
Not for this PR, but I wonder if it may make sense to make `tokio` an optional dependency - i.e., maybe the user of the rate-limiter could control how the cleanup is invoked. Let's create an issue for this.